model/perf.go

```diff
@@ -345,6 +345,12 @@ var (
 type PerformanceArguments map[string]int32
 
 func (args PerformanceArguments) MarshalBSON() ([]byte, error) {
+	// If the arguments map is empty or only has one item, we can just use
+	// the default marshaler.
+	if len(args) < 2 {
+		return bson.Marshal(map[string]int32(args))
+	}
+
 	sortedArgs := bson.D{}
```

Comment:
would an alternative be to 
```
var sortedArgs bson.D
```
My thinking is that this way sortedArgs will be nil (unless it's appended to) and get marshaled to null.

Reply:
I believe, in that case, it would have to be `var sortedArgs *bson.D`, `var sortedArgs bson.D` would still marshal to `{}`.

Reply:
Actually, `bson.D` wraps a slice. So this would work.